article/_posts/2020-11-19-AlphaMuGoZero.md

```diff
@@ -0,0 +1,181 @@
+---
+layout: review
+title: "AlphaGo/AlphaGoZero/AlphaZero/MuZero: Mastering games using progressively fewer priors"
+tags: deep-learning reinforcement
+author: "Antoine Théberge"
+cite:
+    authors: "Julian Schrittwieser, Ioannis Antonoglou, Thomas Hubert, Karen Simonyan, Laurent Sifre, Simon Schmitt, Arthur Guez, Edward Lockhart, Demis Hassabis, Thore Graepel, Timothy Lillicrap, David Silver"
+    title:   "Mastering Atari, Go, Chess and Shogi by Planning with a Learned Model"
+    venue:   "arXiv"
+pdf: ""
+---
+
+
+# Highlights
+
+- AlphaGo defeated the Go champion Lee Sedol in 2016 in a best-of-five tournament
+- Since then, Deepmind has released three subsequent versions of the algorithm, each imposing fewer priors and increasing performance
+
+# Introduction
+
+In 2016, Deepmind famously put AlphaGo against Go world-champion Lee Sedol and won four matches out of five in tournament. Surprisingly, the algorithm used both recent advances in supervised machine learning and deep reinforcement learning as well as the classical heuristic search algorithm Monte Carlo tree search.
+
+## Monte Carlo Tree Search
+
+![](/article/images/alpha/mcts.jpg)
+> source: wikipedia.org
+
+Monte Carlo tree search (MCTS) was actually first introduced in 2006 to play the game of Go[^1] and won the 10th computer-Go tournament. It has since then been applied to many problems including chess, shogi, bridge, poker, and video games such as Starcraft.
+
+As the name implies, MCTS is a tree search algorithm that uses Monte Carlo sampling of the search space to iteratively refine the evaluation of nodes. From a root node, child nodes are traversed until a leaf node (a node that has one or more children that has not been evaluated or until a maximum depth is reached) is reached. Then, a child is selected and a rollout (selecting actions according to a possibly random policy until the end of the game) is performed, assigning the result to the node. Finally, the node's value is "backpropagated" to parent nodes. Finally, the optimal action can be inferred from the node's values. This process is repeated for each move.
```

Comment:
Not sure if this sentence requires being re-worded: "From a root node, child nodes are traversed until a leaf node (a node that has one or more children that has not been evaluated or until a maximum depth is reached) is reached."

Can a leaf node have children?

Reply:
Reworded in 137ef52